### ISSUE TITLE
[v1.2-0-beta.0] Sorting concordances duplicates collocate highlighting

### ISSUE THREAD

**Issue by violawiegand** [birmingham-ccr/CorporaCoCo#3]
This issue concerns a problem with sorting the output from the `corp_concordance()` function when collocates are supplied.

# Test data

I tested this with the [raw text of *Hard Times* retrieved from the Corpora repository](https://github.com/birmingham-ccr/corpora/blob/f87af1d2c9489cc7cd1fd29f061204317442a914/DNov/HT.txt), turning it into a corp_text object as follows (same as I did with [this issue](https://github.com/birmingham-ccr/CorporaCoCo/issues/2)):

```
a <- readLines("HT.txt")
a_text_collapsed <- paste(a, collapse = "")
a_text <- corp_text(a_text_collapsed)
```

# Running a concordance & adding "collocates"

If I run a concordance for "hands" and set the "collocates" to "his",

```
y <- corp_concordance(a_text, span = "5LR", nodes = c("hands"), collocates = "his")
y
```
I retrieve the following output, with "his" highlighted when it occurs in the specified span (as intended):
```
[ 1] bringing up.' 'Whether,' --- said Gradgrind, pondering with *his* hands in *his* pockets, and *his* --- cavernous eyes on          
[ 2]                 fashion of wearing --- *his* hat - and with *his* hands in *his* pockets, sauntered out --- into the hall          
[ 3]              By your leaves, --- gentlemen!' walked in with *his* hands in *his* pockets. *His* face, --- close-shaven, thin       
[ 4]                           E. W. B. --- Childers took one of *his* hands out of *his* pockets, stroked --- his face and             
[ 5]                      hair all manner --- of ways with *his* sulky hands, 'I am a Donkey, that's --- what I am                      
[ 6]                        of it, with --- *his* face hidden in *his* hands. 'Come awa! from 't. 'Tis --- mine, and I've               
[ 7]                        throw a covering --- over her; as if *his* hands were not enough to hide --- her, even in                   
[ 8]             tell you what!' --- said Mr. Bounderby, putting *his* hands in *his* pockets. 'There is --- such a law                 
[ 9]                   so far off, --- but that Stephen, following her hands with *his* eyes, could read --- what was printed           
[10]                            or her, or --- both!"' She put her two hands on *his* mouth, with a --- face of terror                  
[11]                secret place, his --- work is noiseless, and *his* Hands are mutes. CHAPTER XV. - FATHER --- AND DAUGHTERALTHOUGH Mr
[12]       Mrs. Sparsit, ma'am,' --- said Mr. Bounderby, putting *his* hands in *his* pockets, and assuring --- himself with his        
[13]                  functionary now took --- the case into *his* own hands. 'My friends,' said he, 'by --- virtue o' my               
[14]                            be so. I --- don't know.' She took her hands suddenly from *his* shoulders, and --- pressed them both   
[15]        Tom Gradgrind,' said --- Bounderby, standing up with *his* hands in *his* pockets, 'that you --- are of opinion             
[16]             confronting him with --- *his* legs wide apart, *his* hands deeper in *his* pockets, and --- his hair like             
[17]       that's true enough,' --- assented Mr. Bounderby, with *his* hands in *his* pockets and *his* --- hat on. 'But                
[18]               and gone away --- again.' Mr. Bounderby, with *his* hands in *his* pockets, walked in --- impatient mortification up 
[19]              yourself!' The father --- buried *his* face in *his* hands, and the son stood in --- his disgraceful grotesqueness    
[20]   stood in his --- disgraceful grotesqueness, biting straw: *his* hands, with the black partly worn --- away inside, looking       
[21]     Bank.' 'Bitzer,' said --- Mr. Gradgrind, stretching out *his* hands as though he would have --- said, See how 
```
# Accumulating asterisks when sorting concordances
When I then sort the same concordance by the first type on the right, a problem occurs with the highlighting. The sorting now creates double asterisks, although it should just retain the single asterisks. So we would still want to see `*his*` rather than the retrieved `**his**`:

```{r sorting}
y[order(N, R1_type)]
```

```
[ 1]                secret place, his --- work is noiseless, and **his** Hands are mutes. CHAPTER XV. - FATHER --- AND DAUGHTERALTHOUGH Mr 
[ 2]            yourself!' The father --- buried **his** face in **his** hands, and the son stood in --- his disgraceful grotesqueness     
[ 3]     Bank.' 'Bitzer,' said --- Mr. Gradgrind, stretching out **his** hands as though he would have --- said, See how                   
[ 4]                      of it, with --- **his** face hidden in **his** hands. 'Come awa! from 't. 'Tis --- mine, and I've                
[ 5]           confronting him with --- **his** legs wide apart, **his** hands deeper in **his** pockets, and --- his hair like            
[ 6]                      hair all manner --- of ways with **his** sulky hands, 'I am a Donkey, that's --- what I am                       
[ 7] bringing up.' 'Whether,' --- said Gradgrind, pondering with **his** hands in **his** pockets, and **his** --- cavernous eyes on       
[ 8]               fashion of wearing --- **his** hat - and with **his** hands in **his** pockets, sauntered out --- into the hall         
[ 9]              By your leaves, --- gentlemen!' walked in with **his** hands in **his** pockets. **His** face, --- close-shaven, thin    
[10]             tell you what!' --- said Mr. Bounderby, putting **his** hands in **his** pockets. 'There is --- such a law                
[11]       Mrs. Sparsit, ma'am,' --- said Mr. Bounderby, putting **his** hands in **his** pockets, and assuring --- himself with his       
[12]        Tom Gradgrind,' said --- Bounderby, standing up with **his** hands in **his** pockets, 'that you --- are of opinion            
[13]       that's true enough,' --- assented Mr. Bounderby, with **his** hands in **his** pockets and **his** --- hat on. 'But             
[14]               and gone away --- again.' Mr. Bounderby, with **his** hands in **his** pockets, walked in --- impatient mortification up
[15]                  functionary now took --- the case into **his** own hands. 'My friends,' said he, 'by --- virtue o' my                
[16]                              or her, or --- both!"' She put her two hands on **his** mouth, with a --- face of terror                 
[17]                           E. W. B. --- Childers took one of **his** hands out of **his** pockets, stroked --- his face and            
[18]                              be so. I --- don't know.' She took her hands suddenly from **his** shoulders, and --- pressed them both  
[19]                        throw a covering --- over her; as if **his** hands were not enough to hide --- her, even in                    
[20]                     so far off, --- but that Stephen, following her hands with **his** eyes, could read --- what was printed          
[21]   stood in his --- disgraceful grotesqueness, biting straw: **his** hands, with the black partly worn --- away inside, looking        
```

It seems the number of asterisks increases when you sort the concordance multiple times on the same side, because if I sort it again on the right (in this case for another "R" position, but the same happens if I sort by R1_type again), I now get three asterisks for highlighting:

```{r sorting_2}
y[order(N, R2_type)]
y
```
```
[ 1]                secret place, his --- work is noiseless, and ***his*** Hands are mutes. CHAPTER XV. - FATHER --- AND DAUGHTERALTHOUGH Mr   
[ 2]                      hair all manner --- of ways with ***his*** sulky hands, 'I am a Donkey, that's --- what I am                         
[ 3]                    of it, with --- ***his*** face hidden in ***his*** hands. 'Come awa! from 't. 'Tis --- mine, and I've                  
[ 4]                  functionary now took --- the case into ***his*** own hands. 'My friends,' said he, 'by --- virtue o' my                  
[ 5]                                be so. I --- don't know.' She took her hands suddenly from ***his*** shoulders, and --- pressed them both  
[ 6] bringing up.' 'Whether,' --- said Gradgrind, pondering with ***his*** hands in ***his*** pockets, and ***his*** --- cavernous eyes on     
[ 7]             fashion of wearing --- ***his*** hat - and with ***his*** hands in ***his*** pockets, sauntered out --- into the hall         
[ 8]              By your leaves, --- gentlemen!' walked in with ***his*** hands in ***his*** pockets. ***His*** face, --- close-shaven, thin  
[ 9]             tell you what!' --- said Mr. Bounderby, putting ***his*** hands in ***his*** pockets. 'There is --- such a law                
[10]                       so far off, --- but that Stephen, following her hands with ***his*** eyes, could read --- what was printed          
[11]                                or her, or --- both!"' She put her two hands on ***his*** mouth, with a --- face of terror                 
[12]       Mrs. Sparsit, ma'am,' --- said Mr. Bounderby, putting ***his*** hands in ***his*** pockets, and assuring --- himself with his       
[13]        Tom Gradgrind,' said --- Bounderby, standing up with ***his*** hands in ***his*** pockets, 'that you --- are of opinion            
[14]       that's true enough,' --- assented Mr. Bounderby, with ***his*** hands in ***his*** pockets and ***his*** --- hat on. 'But           
[15]               and gone away --- again.' Mr. Bounderby, with ***his*** hands in ***his*** pockets, walked in --- impatient mortification up
[16]         confronting him with --- ***his*** legs wide apart, ***his*** hands deeper in ***his*** pockets, and --- his hair like            
[17]                        throw a covering --- over her; as if ***his*** hands were not enough to hide --- her, even in                      
[18]                           E. W. B. --- Childers took one of ***his*** hands out of ***his*** pockets, stroked --- his face and            
[19]          yourself!' The father --- buried ***his*** face in ***his*** hands, and the son stood in --- his disgraceful grotesqueness       
[20]   stood in his --- disgraceful grotesqueness, biting straw: ***his*** hands, with the black partly worn --- away inside, looking          
[21]     Bank.' 'Bitzer,' said --- Mr. Gradgrind, stretching out ***his*** hands as though he would have --- said, See how
```
The asterisks **do not** increase if I switch from sorting on the right to sorting on the left.

 
I encountered this issue with the following versions of R, RStudio, packages etc.

platform       x86_64-apple-darwin15.6.0   
arch           x86_64                      
os             darwin15.6.0                
system         x86_64, darwin15.6.0        
status                                     
major          3                           
minor          6.1                         
year           2019                        
month          07                          
day            05                          
svn rev        76782                       
language       R                           
version.string R version 3.6.1 (2019-07-05)
nickname       Action of the Toes     

RStudio version 1.1.463
CorporaCoCo version v1.2-0-beta.0 
data.table version 1.12.8

**Comment by MHenderson** [comment]
This also happens when you print `y` more than once. The reason is that the `print.corp_concordance` function actually updates the underlying data by adding asterisks to the relevant columns in the data. The simplest solution I could think of was just to copy the underlying data before updating it for the purpose of printing it out. It solves this issue but I can't be certain that it won't throw up some other issue elsewhere.

**Comment by edmondac** [comment]
Fixed by #6
